Reorganize imgpkg Docs, Write Workflow Docs, and Add imgpkg tag list Docs (#21)
carvel-dev/imgpkg: docs/README.md

# imgpkg Documentation

### What is imgpkg

`imgpkg` is a tool that allows users to store and distribute sets of files as OCI images.
A typical use for these OCI images is to group configurations for a particular application 
and make it available in an image registry.

The tool introduces the concept of a Bundle, which is an OCI image that contains configuration files and
references of images that can be used with these configurations.

Currently, `imgpkg` always produces a single layer image. It's not optimized to repush 
large sized directories that infrequently change.

### Images vs Bundles

An image contains a generic set of files or directories. Ultimately, an image is a tarball of all the provided inputs.

A bundle is an image with some additional characteristics:
- Contains a bundle directory (`.imgpkg/`), which must exist at the root-level of the bundle and
  contain info about the bundle, such as an [ImagesLock](resources.md#imageslock) and,
  optionally, a [bundle metadata file](resources.md#bundle-metadata)

> - Has a config label notating that the image is a bundle

Reviewer: ```suggestion
- Has the `dev.carvel.imgpkg.bundle` [label](https://docs.docker.com/config/labels-custom-metadata/) marking the image as an imgpkg Bundle
```

Author: Thanks, added.
